### DISCUSSION TITLE
Responsive PNG doesn't seem optimal

### DISCUSSION THREAD

**Discussion by stuartcusackie** [spatie/laravel-medialibrary#3271] [Ideas]
Hi. I am using the withResponsiveImages() method and the default with calculator on a transparent PNG, which is 1000 pixels wide.

The markup that is being produced doesn't seem very well optimised. Only three sizes are generated with the smallest being 699 pixels wide and 20kb:

`<img srcset="https://ams3.digitaloceanspaces.com/spacenamex/media/8/responsive-images/image1test___default_1000_281.png 1000w, https://ams3.digitaloceanspaces.com/spacenamex/media/8/responsive-images/image1test___default_836_235.png 836w, https://ams3.digitaloceanspaces.com/spacenamex/media/8/responsive-images/image1test___default_699_196.png 699w, data:image/svg+xml;base64,{REDACATEDSVGDATA} 32w" onload="window.requestAnimationFrame(function(){if(!(size=getBoundingClientRect().width))return;onload=null;sizes=Math.ceil(size/window.innerWidth*100)+'vw';});" sizes="20vw" src="https://ams3.digitaloceanspaces.com/spacenamex/media/8/conversions/image1test-default.png" width="1000" height="281">`

Here's my Model definition:
```
 public function registerMediaCollections(): void
    {
        $this
            ->addMediaCollection('default')
            ->useFallbackUrl('/assets/images/standalone-fallback.jpg')
            ->useFallbackPath(public_path('/assets/images/standalone-fallback.jpg'))
            ->singleFile()
            ->registerMediaConversions(function (Media $media) {

                $this
                    ->addMediaConversion('default')
                    ->keepOriginalImageFormat()
                    ->withResponsiveImages()
                    ->nonQueued();
                    
            });
    }
```

My Jpegs seem fine and much smaller variations are generated.

**Comment by freekmurze** [comment]
Feel free to PR improvements around this!
